src/common/src/utils.rs

```diff
@@ -12,3 +12,35 @@ pub fn init_nones<T, A: FixedSizeArray<Option<T>>>() -> A {
     res
 }
 
+/// Allocates a global array of static size, initially set to zero. When this
+/// macro is hit, it will initialize the array to the value given and return a
+/// `&'static mut` reference to it.
+///
+/// Note that you will have to specify the array-size as an argument, but a
+/// wrong size will result in a compile-time error. This argument will be
+/// removed if `std::mem::size_of<T>` ever becomes a `const` function.
```

Comment:
I will update with the requirement for unsafe. However there is no problem with calling this macro twice -- it will just allocate two areas.